# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1241544735 14400
#      Tue May 05 13:32:15 2009 -0400
# Node ID 86ae7e50dc5dea037d4121988f04b42b1c54990e
# Parent  2669527e0ce58003b3da1d920793a74d39894d16
graphics.cc (array_property::validate): require value to be numeric, not necessarily double

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,13 @@
 2009-05-05  John W. Eaton  <jwe@octave.org>
 
+	* graphics.cc (array_property::validate): Require object to be any
+	numeric type, not necessarily a double precision value.
+
 	* variables.cc (set_internal_variable): Pass NM in call to error.
 
 2009-05-04  Peter O'Gorman  <pogma@thewrittenword.com>
 
 	* utils.cc: Don't define HAVE_C99_VSNPRINTF here.
 
 2009-05-01  John W. Eaton  <jwe@octave.org>
 
diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -771,29 +771,29 @@ double_radio_property::do_set (const oct
 }
 
 bool
 array_property::validate (const octave_value& v)
 {
   bool xok = false;
 
   // FIXME -- should we always support []?
-  if (v.is_empty () && v.is_double_type ())
+  if (v.is_empty () && v.is_numeric_type ())
     return true;
 
   // check value type
   if (type_constraints.size () > 0)
     {
       for (std::list<std::string>::const_iterator it = type_constraints.begin ();
            ! xok && it != type_constraints.end (); ++it)
         if ((*it) == v.class_name ())
           xok = true;
     }
   else
-    xok = v.is_double_type ();
+    xok = v.is_numeric_type ();
 
   if (xok)
     {
       dim_vector vdims = v.dims ();
       int vlen = vdims.length ();
 
       xok = false;
 
